File: 0066_Plus One/PlusOne.cs
public class Solution {
    public int[] PlusOne(int[] digits) {
        if(digits == null || digits.Length == 0) return new int[1]{1};
        
        var stack = new Stack<int>();
        var carry = true;
        
        var index = digits.Length - 1;
        
        while(index>=0 || carry){
            
            var num = carry ? 1 : 0;
            if(index >= 0){
                num += digits[index--];
            }
            
            carry = num >= 10;
            num %= 10;
            
            stack.Push(num);
            
        }
        
        var ans = new int[stack.Count];
        for(int i=0;i<ans.Length;i++){
            ans[i] = stack.Pop();
        }
        
        return ans;
    }
}

File: 0066_Plus One/PlusOne.java
class Solution {
    public int[] plusOne(int[] digits) {
        int n = digits.length, carry = 1;
        for(int i=n-1;i>=0;i--) {
            int v = carry + digits[i];
            carry = v / 10;
            digits[i] = v % 10;
        }
        
        if(carry > 0) {
            int[] ans = new int[n+1];
            ans[0] = carry;
            for(int i=0;i<n;i++) {
                ans[i+1] = digits[i];
                return ans;
            }
        }
        
        return digits;
    }
}